source/ingress.go

```diff
@@ -129,6 +129,11 @@ func (sc *ingressSource) Endpoints(ctx context.Context) ([]*endpoint.Endpoint, e
 
 		ingEndpoints := endpointsFromIngress(ing, sc.ignoreHostnameAnnotation, sc.ignoreIngressTLSSpec, sc.ignoreIngressRulesSpec)
 
+		// Template configurations are a fallback to determine endpoint targets
+		if sc.fqdnTemplate != nil && endpoint.EndpointsHaveEmptyTargets(ingEndpoints) {
+			ingEndpoints = []*endpoint.Endpoint{}
```

Comment:
It looks to me that if a fqdn-template is provided, then endpoints with empty targets will be removed (i.e. deleted from the provider). Why the special case here?

Reply:
If a template is provided, it will be used instead of removing the endpoint.
If this isn't done templates would be ignored every time.

Reply:
I'm seeing it still be deleted.

As it stands, if I provide a fqdn-template, it calls [endpointsFromTemplate](https://github.com/Jimdo/external-dns/blob/master/source/ingress.go#L171) which looks up the targets the same way from the ingress (target annotation override, else ingress status) and passes those targets to the common `source.endpointsForHostname` where you end up with an empty endpoint list and that will cause a deletion if the endpoint had a DNS record created prior.

Reply:
Oh, you are correct. I somehow missed `endpointsFromTemplate` calling `endpointsForHostname` too.